homeassistant/components/pegel_online/sensor.py

```diff
@@ -24,67 +25,67 @@
 class PegelOnlineSensorEntityDescription(SensorEntityDescription):
     """PEGELONLINE sensor entity description."""
 
-    measurement_key: str
+    measurement_fn: Callable[[StationMeasurements], CurrentMeasurement | None]
 
 
 SENSORS: tuple[PegelOnlineSensorEntityDescription, ...] = (
     PegelOnlineSensorEntityDescription(
         key="air_temperature",
         translation_key="air_temperature",
-        measurement_key="air_temperature",
         state_class=SensorStateClass.MEASUREMENT,
         device_class=SensorDeviceClass.TEMPERATURE,
         entity_registry_enabled_default=False,
+        measurement_fn=lambda data: data.air_temperature,
     ),
     PegelOnlineSensorEntityDescription(
         key="clearance_height",
         translation_key="clearance_height",
-        measurement_key="clearance_height",
         state_class=SensorStateClass.MEASUREMENT,
         device_class=SensorDeviceClass.DISTANCE,
+        measurement_fn=lambda data: data.clearance_height,
     ),
     PegelOnlineSensorEntityDescription(
         key="oxygen_level",
         translation_key="oxygen_level",
-        measurement_key="oxygen_level",
         state_class=SensorStateClass.MEASUREMENT,
         entity_registry_enabled_default=False,
+        measurement_fn=lambda data: data.oxygen_level,
     ),
     PegelOnlineSensorEntityDescription(
         key="ph_value",
-        measurement_key="ph_value",
         state_class=SensorStateClass.MEASUREMENT,
         device_class=SensorDeviceClass.PH,
         entity_registry_enabled_default=False,
+        measurement_fn=lambda data: data.ph_value,
     ),
     PegelOnlineSensorEntityDescription(
         key="water_speed",
         translation_key="water_speed",
-        measurement_key="water_speed",
         state_class=SensorStateClass.MEASUREMENT,
         device_class=SensorDeviceClass.SPEED,
         entity_registry_enabled_default=False,
+        measurement_fn=lambda data: data.water_speed,
     ),
     PegelOnlineSensorEntityDescription(
         key="water_flow",
```

Comment:
also, should this get the volume flow rate device class?

Reply:
uhh nice ... wan't aware, that we have such class - should I add it in this PR?

Reply:
no, different PR

Reply:
ok ... now I remerber - the api provides the data in `m³/s` but this is not an accepted uom for the `VOLUME_FLOW_RATE` device class

Reply:
will open an arch discussion to extend the device class :+1:

Reply:
there was already an open discussion -> https://github.com/home-assistant/architecture/discussions/1044

Reply:
done -> #143631